arch/x86/core/intel64/locore.S

```diff
@@ -820,16 +820,18 @@ IRQ(248); IRQ(249); IRQ(250); IRQ(251); IRQ(252); IRQ(253); IRQ(254); IRQ(255)
 #define	IRQ_STACK	2
 #define EXC_STACK	2
 #define BAD_STACK	2
+#define NMI_STACK	2
 #else
 #define	IRQ_STACK	1
+#define NMI_STACK	6 /* NMI stack */
 #define EXC_STACK	7
-#define BAD_STACK	7 /* Horrible things: NMIs, double faults, MCEs */
+#define BAD_STACK	7 /* Horrible things: double faults, MCEs */
```

Comment:
is it possible to use the NMI stack for these as well? can an MCE fire while an NMI is being handled? a double fault is always a pathological error.

Reply:
Do you mean make BAD_STACK = NMI_STACK = 6 ? I can change to this.
Then you asked to name arrays as `z_x86_nmi_stackN`, probably I need better name.